specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+            <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          )}
+        </>
+      }
+      dimensionsKey="ModifyUniquenessRule"
+      header={schemaText.configureUniquenessRule()}
+      icon={icons.pencilAt}
+      modal
+      onClose={handleClose}
+    >
+      <>
+        <H2>{label}</H2>
+        <p>{schemaText.uniqueFields()}</p>
+        {rule.fields.map((field, index) => (
+          <div className="inline-flex" key={index}>
+            <PickList
+              disabled={readOnly}
+              groups={{
+                [schemaText.fields()]: uniqueFields,
+              }}
+              value={field}
+              onChange={(value): void => {
+                const newField =
+                  fields.find(({ name }) => name === value) ??
+                  relationships.find(({ name }) => name === value);
+                if (newField === undefined) return;
+                handleChanged({
+                  ...rule,
+                  fields: replaceItem(rule.fields, index, newField.name),
+                });
+              }}
+            />
+            {rule.fields.length > 1 && (
+              <Button.Icon
+                className={`w-fit ${className.dataEntryRemove}`}
+                disabled={readOnly}
+                icon="minus"
+                title={commonText.remove()}
+                onClick={(): void =>
+                  handleChanged({
+                    ...rule,
+                    fields: removeItem(rule.fields, index),
+                  })
+                }
+              />
+            )}
+          </div>
+        ))}
+        <Button.BorderedGray
+          className="w-fit"
+          disabled={readOnly}
+          onClick={(): void =>
+            handleChanged({
+              ...rule,
+              fields: insertItem(
+                rule.fields,
+                rule.fields.length,
+                fields.find(({ name }) => !rule.fields.includes(name))!.name
+              ),
+            })
+          }
+        >
+          {commonText.add()}
+        </Button.BorderedGray>
+        <p>{schemaText.scope()}</p>
+        <UniquenessRuleScope
+          model={model}
+          rule={rule}
+          onChange={handleChanged}
+        />
+        <Input.Text
+          disabled
+          value={
+            rule.scopes.length === 0
+              ? schemaText.database()
+              : getFieldsFromPath(model, rule.scopes[0])
+                  .map((field) => field.localization.name as string)
+                  .join(' -> ')
+          }
+        />
+      </>
+    </Dialog>
+  );
+}
+
+function UniquenessRuleScope({
+  rule,
+  model,
+  onChange: handleChanged,
+}: {
+  readonly rule: UniquenessRule;
+  readonly model: SpecifyModel;
+  readonly onChange: (newRule: typeof rule) => void;
+}): JSX.Element {
+  const [mappingPath, setMappingPath] = React.useState<RA<string>>(
+    rule.scopes.length === 0 ? ['database'] : rule.scopes[0].split('__')
```

Comment:
where is `__` coming from? (i.e what code decides that __ is used to join the string)
could it be extracted into a constant and reused from there to make it easy to find all the usages?

Reply:
okay - then could you please define a variable on the front-end that would equal the `__`
and then use the variable everywhere you used `__` on the front-end before

benefits:
- easy to find all the usages once it's a variable
- in the place where you declared the variable, could optionally attach a small comment explaining what this variable is needed for (like the kind of explanation [you sent above](https://github.com/specify/specify7/pull/4215/files/6acf0c6283c0513cf956d45223b4024a99ff02f4..43aee5b97e52edca55b58e0cb2b76773754b510f#r1443309543))
- reduces possibility of accidentally making a typo (a typo in a string is not caught by TypeScript, a typo in a variable name is caught by TypeScript)